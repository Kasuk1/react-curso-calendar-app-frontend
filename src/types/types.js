export const types = {
  uiOpenModal: '[ui] Open modal',
  uiCloseModal: '[ui] Close modal',

  authCheckingFinish: '[auth] Finish checking login state',
  authStartLogin: '[auth] Start login',
  authLogin: '[auth] Login',
  authStartRegister: '[auth] Start register',
  authStartTokenRenew: '[auth] Start token renew',
  authLogout: '[auth] Logout',

  eventSetActive: '[event] Set active',
  eventLogout: '[event] Logout',
  eventAddNew: '[event] Add new',
  eventStartAddNew: '[event] Start add new',
  eventClearActiveEvent: '[event] Clear active event',
  eventUpdated: '[event] Update event',
  eventDeleted: '[event] Delete event',
  eventLoaded: '[event] Event loaded',
};
