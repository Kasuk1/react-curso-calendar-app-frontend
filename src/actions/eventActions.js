import Swal from 'sweetalert2';
import { fetchWithToken } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';
import { types } from '../types/types';

export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;
    try {
      const resp = await fetchWithToken('events', event, 'POST');
      const json = await resp.json();
      if (json.ok) {
        event.id = json.event.id;
        event.user = {
          _id: uid,
          name,
        };
        dispatch(eventAddNew(event));
      }
    } catch (err) {
      console.log(err);
    }
  };
};

const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event,
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event,
});

export const eventClearActiveEvent = () => ({
  type: types.eventClearActiveEvent,
});

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken(`events/${event.id}`, event, 'PUT');
      const json = await resp.json();

      if (json.ok) {
        dispatch(eventUpdated(event));
      } else {
        Swal.fire('Error', json.msg, 'error');
      }
    } catch (err) {
      console.log(err);
    }
  };
};

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event,
});

export const eventStartDelete = (eventId) => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken(`events/${eventId}`, {}, 'DELETE');
      const json = await resp.json();
      if (json.ok) {
        dispatch(eventDeleted());
      } else {
        Swal.fire('Error', json.msg, 'error');
      }
    } catch (err) {
      console.log(err);
    }
  };
};

const eventDeleted = () => ({
  type: types.eventDeleted,
});

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken('events');
      const json = await resp.json();
      const events = prepareEvents(json.events);

      if (json.ok) {
        dispatch(eventLoaded(events));
      }
    } catch (err) {
      console.log(err);
    }
  };
};

const eventLoaded = (events) => ({
  type: types.eventLoaded,
  payload: events,
});

export const eventLogout = () => ({
  type: types.eventLogout,
});
