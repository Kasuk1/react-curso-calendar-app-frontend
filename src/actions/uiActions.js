import { types } from '../types/types';

export const openModal = () => ({
  type: types.uiOpenModal,
});

export const closeModal = () => ({
  type: types.uiCloseModal,
});
