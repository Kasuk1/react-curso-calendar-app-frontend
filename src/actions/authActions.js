import Swal from 'sweetalert2';
import { fetchNoToken, fetchWithToken } from '../helpers/fetch';
import { types } from '../types/types';
import { eventLogout } from './eventActions';

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const resp = await fetchNoToken('auth', { email, password }, 'POST');
    const json = await resp.json();

    if (json.ok) {
      localStorage.setItem('token', json.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      dispatch(
        login({
          uid: json.uid,
          name: json.name,
        })
      );
    } else {
      Swal.fire('Error', json.msg, 'error');
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    const resp = await fetchNoToken(
      'auth/new',
      { name, email, password },
      'POST'
    );
    const json = await resp.json();

    if (json.ok) {
      localStorage.setItem('token', json.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      dispatch(
        login({
          uid: json.uid,
          name: json.name,
        })
      );
    } else {
      Swal.fire('Error', json.msg, 'error');
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchWithToken('auth/renew');
    const json = await resp.json();

    if (json.ok) {
      localStorage.setItem('token', json.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      dispatch(
        login({
          uid: json.uid,
          name: json.name,
        })
      );
    } else {
      dispatch(checkingFinish());
    }
  };
};

const login = (user) => ({
  type: types.authLogin,
  payload: user,
});

const checkingFinish = () => ({
  type: types.authCheckingFinish,
});

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(logout());
    dispatch(eventLogout());
  };
};

const logout = () => ({
  type: types.authLogout,
});
