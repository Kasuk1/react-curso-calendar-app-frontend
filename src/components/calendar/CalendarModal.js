import React, { useEffect, useState } from 'react';
import Modal from 'react-modal';
import DateTimePicker from 'react-datetime-picker';
import moment from 'moment';
import Swal from 'sweetalert2';
import { useDispatch, useSelector } from 'react-redux';
import { closeModal } from '../../actions/uiActions';
import {
  eventClearActiveEvent,
  eventStartAddNew,
  eventStartUpdate,
} from '../../actions/eventActions';

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
};

if (process.env.NODE_ENV !== 'test') {
  Modal.setAppElement('#root');
}

const now = moment().minutes(0).seconds(0).add(1, 'hours');
const nowPlusOne = now.clone().add(1, 'hours');

const initEvent = {
  title: '',
  notes: '',
  start: now.toDate(),
  end: nowPlusOne.toDate(),
};

export const CalendarModal = () => {
  const dispatch = useDispatch();
  const { uid: currentUserId } = useSelector((state) => state.auth);
  const { modalOpen } = useSelector((state) => state.ui);
  const { activeEvent } = useSelector((state) => state.calendar);
  const [dateStart, setDateStart] = useState(now.toDate());
  const [dateEnd, setDateEnd] = useState(nowPlusOne.toDate());
  const [titleValid, setTitleValid] = useState(true);

  const [formValues, setFormValues] = useState(initEvent);

  const { title, notes, start, end } = formValues;

  useEffect(() => {
    if (activeEvent) {
      setFormValues(activeEvent);
      setDateStart(activeEvent.start);
      setDateEnd(activeEvent.end);
    } else {
      setFormValues(initEvent);
    }
  }, [activeEvent]);

  const handleInputChange = ({ target }) => {
    setFormValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  const handleCloseModal = () => {
    dispatch(closeModal());
    dispatch(eventClearActiveEvent());
    setFormValues(initEvent);
  };

  const handleStartDateChange = (e) => {
    setDateStart(e);
    setFormValues({
      ...formValues,
      start: e,
    });
  };

  const handleEndDateChange = (e) => {
    setDateEnd(e);
    setFormValues({
      ...formValues,
      end: e,
    });
  };

  const handleSubmitForm = (e) => {
    e.preventDefault();

    const momentStart = moment(start);
    const momentEnd = moment(end);

    if (momentStart.isSameOrAfter(momentEnd)) {
      return Swal.fire(
        'Error',
        'La fecha fin debe ser mayor a la de inicio',
        'error'
      );
    }

    if (title.trim().length < 2) {
      return setTitleValid(false);
    }

    if (activeEvent) {
      dispatch(eventStartUpdate(formValues));
    } else {
      dispatch(eventStartAddNew(formValues));
    }
    setTitleValid(true);
    handleCloseModal();
  };

  return (
    <Modal
      className='modal'
      isOpen={modalOpen}
      ///onAfterOpen={afterOpenModal}
      onRequestClose={handleCloseModal}
      style={customStyles}
      closeTimeoutMS={200}
      overlayClassName='modal-fondo'
      ariaHideApp={!process.env.NODE_ENV === 'test'}
    >
      <h1> {activeEvent ? 'Editar evento' : 'Nuevo evento'} </h1>
      <hr />
      <form className='container' onSubmit={handleSubmitForm}>
        <div className='form-group'>
          <label>Fecha y hora inicio</label>
          <DateTimePicker
            onChange={handleStartDateChange}
            value={dateStart}
            className='form-control'
          />
        </div>

        <div className='form-group'>
          <label>Fecha y hora fin</label>
          <DateTimePicker
            onChange={handleEndDateChange}
            value={dateEnd}
            minDate={dateStart}
            className='form-control'
          />
        </div>

        <hr />
        <div className='form-group'>
          <label>Titulo y notas</label>
          <input
            type='text'
            className={`form-control ${!titleValid && 'is-invalid'}`}
            placeholder='Título del evento'
            name='title'
            autoComplete='off'
            value={title}
            onChange={handleInputChange}
          />
          <small id='emailHelp' className='form-text text-muted'>
            Una descripción corta
          </small>
        </div>

        <div className='form-group'>
          <textarea
            type='text'
            className='form-control'
            placeholder='Notas'
            rows='5'
            name='notes'
            value={notes}
            onChange={handleInputChange}
          ></textarea>
          <small id='emailHelp' className='form-text text-muted'>
            Información adicional
          </small>
        </div>

        {activeEvent?.user._id === currentUserId ||
          (!activeEvent && (
            <button type='submit' className='btn btn-outline-primary btn-block'>
              <i className='far fa-save'></i>
              <span> Guardar</span>
            </button>
          ))}
      </form>
    </Modal>
  );
};
