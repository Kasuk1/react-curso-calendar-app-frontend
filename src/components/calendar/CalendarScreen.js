import React, { useEffect, useState } from 'react';
import { Calendar, momentLocalizer } from 'react-big-calendar';
import moment from 'moment';
import 'moment/locale/es';
import 'react-big-calendar/lib/css/react-big-calendar.css';

import { Navbar } from '../ui/Navbar';
import { messages } from '../../helpers/calendar-messages-espaÃ±ol';
import { CalendarEvent } from './CalendarEvent';
import { CalendarModal } from './CalendarModal';
import { useDispatch, useSelector } from 'react-redux';
import { openModal } from '../../actions/uiActions';
import {
  eventClearActiveEvent,
  eventSetActive,
  eventStartLoading,
} from '../../actions/eventActions';
import { AddNewFab } from '../ui/AddNewFab';
import { DeleteEventFab } from '../ui/DeleteEventFab';

moment.locale('es');

const localizer = momentLocalizer(moment);

export const CalendarScreen = () => {
  const dispatch = useDispatch();
  const { uid: currentUserId } = useSelector((state) => state.auth);
  const { events, activeEvent } = useSelector((state) => state.calendar);
  const [lastView, setLastView] = useState(
    window.localStorage.getItem('lastView') || 'month'
  );

  useEffect(() => {
    dispatch(eventStartLoading());
  }, [dispatch]);

  const onDoubleClick = (e) => {
    dispatch(openModal());
  };

  const onSelectEvent = (e) => {
    dispatch(eventSetActive(e));
  };

  const onViewChange = (e) => {
    setLastView(e);
    window.localStorage.setItem('lastView', e);
  };

  const onSelectSlot = (e) => {
    dispatch(eventClearActiveEvent());
  };

  const eventStyleGetter = (event, start, end, isSelected) => {
    const {
      user: { _id: eventUserId },
    } = event;
    const style = {
      backgroundColor: eventUserId === currentUserId ? 'orange' : 'gray',
      borderRadius: '0px',
      opacity: 0.8,
      display: 'block',
      color: '#fff',
    };

    return {
      style,
    };
  };

  return (
    <div className='calendar-screen'>
      <Navbar />

      <Calendar
        localizer={localizer}
        events={events}
        startAccessor='start'
        endAccessor='end'
        messages={messages}
        eventPropGetter={eventStyleGetter}
        onDoubleClickEvent={onDoubleClick}
        onSelectEvent={onSelectEvent}
        onView={onViewChange}
        view={lastView}
        onSelectSlot={onSelectSlot}
        selectable={true}
        components={{
          event: CalendarEvent,
        }}
      />

      <AddNewFab />
      {activeEvent && activeEvent?.user._id === currentUserId && (
        <DeleteEventFab />
      )}

      <CalendarModal />
    </div>
  );
};
