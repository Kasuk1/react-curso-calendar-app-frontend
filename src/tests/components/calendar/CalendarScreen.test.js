import { mount } from 'enzyme';
import React from 'react';
import { act } from '@testing-library/react';
import { Provider } from 'react-redux';
import { messages } from '../../../helpers/calendar-messages-espaÃ±ol';

import consfigureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { CalendarScreen } from '../../../components/calendar/CalendarScreen';
import { types } from '../../../types/types';
import { eventSetActive } from '../../../actions/eventActions';

jest.mock('../../../actions/eventActions', () => ({
  eventSetActive: jest.fn(),
  eventStartLoading: jest.fn(),
}));
Storage.prototype.setItem = jest.fn();

const middlewares = [thunk];
const mockStore = consfigureStore(middlewares);

const initState = {
  ui: {
    modalOpen: false,
  },
  auth: {
    uid: '123',
  },
  calendar: {
    events: [],
    activeEvent: {
      user: {
        _id: '1234',
      },
    },
  },
};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <CalendarScreen />
  </Provider>
);

describe('<CalendarScreen /> Tests', () => {
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('tests with calendar interactions', () => {
    const calendar = wrapper.find('Calendar');

    const calendarMessages = calendar.prop('messages');
    expect(calendarMessages).toEqual(messages);

    calendar.prop('onDoubleClickEvent')();
    expect(store.dispatch).toHaveBeenCalledWith({ type: types.uiOpenModal });

    calendar.prop('onSelectEvent')({
      start: 'Hola',
    });
    expect(eventSetActive).toHaveBeenCalledWith({ start: 'Hola' });

    act(() => {
      calendar.prop('onView')('week');
      expect(localStorage.setItem).toHaveBeenCalledWith('lastView', 'week');
    });
  });
});
