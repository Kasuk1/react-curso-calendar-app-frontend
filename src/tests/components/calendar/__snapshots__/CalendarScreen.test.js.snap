// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarScreen /> Tests should render correctly 1`] = `
<div
  className="calendar-screen"
>
  <div
    className="navbar navbar-dark bg-dark mb-4"
  >
    <span
      className="navbar-brand"
    />
    <button
      className="btn btn-outline-danger"
    >
      <i
        className="fa fa-sign-out-alt"
      />
      <span
        onClick={[Function]}
      >
         Salir
      </span>
    </button>
  </div>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &lt;
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &gt;
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        abril 2022
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                28
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                29
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                30
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                31
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                01
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                02
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                03
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                04
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                05
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                06
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                07
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                08
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                09
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                10
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                11
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                12
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                13
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                14
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                15
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                16
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                17
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                18
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                19
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                20
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                21
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                22
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                23
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                24
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                25
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                26
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                27
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                28
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                29
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                30
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                01
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary fab"
    onClick={[Function]}
  >
    <i
      className="fas fa-plus"
    />
  </button>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
</div>
`;
